### ISSUE TITLE
CSS class conflict with bootstrap

### ISSUE THREAD

**Issue by zadam** [excalidraw/excalidraw#6567]
Version: 0.15.2

Hello, I'm integrating Excalidraw into an app which also uses bootstrap. I'm currently looking at a bug which causes the hamburger icon not doing anything upon a click:

![image](https://user-images.githubusercontent.com/617641/236669947-c64ca0aa-75a6-4b2c-bc47-874d8787a15d.png)

Upon closer inspection, it looks to be caused by a class name conflict with bootstrap's "dropdown-menu" class:

![image](https://user-images.githubusercontent.com/617641/236670171-5f28e6c3-a231-4752-810c-e466c2417db2.png)

Basically, the boostrap's styles keep the excalidraw's dropdown hidden. I can fix this locally by forcing a style override, but I guess it would be better to avoid these sort of problems, perhaps by using a vendor prefix for all excalidraw CSS classes?

**Comment by ad1992** [comment]
@zadam in Excalidraw the css is scoped inside `.excalidraw` class so that it never impacts anything outside excalidraw.
 For the above use case it's more of outside css impacting excalidraw (as its not scoped), so you might need to override in those cases using `.excalidraw` class.
```css
.excalidraw {
   .dropdown-menu {
     display: block;
   }
 }
 ```
 However we can surely add a display class in excalidraw css for this particular use case but there could be more impacting if there is a css outside without scoping.

**Comment by zadam** [comment]
@ad1992 Yep, I meant the opposite case - the bootstrap classes are impacting excalidraw. The scoping implemented in excalidraw works only "one way" - protects others, but not itself.

> However we can surely add a display class in excalidraw css for this particular use case but there could be more impacting if there is a css outside without scoping.

This would fix only this specific case, but I think the more general problem of outside classes having conflict with excalidraw would persist. The solution I've seen that would fix it would be prefixing all excalidraw classes with some prefix - e.g. `.exc-dropdown-menu` (e.g. all CKEditor classes start with `.ck-`). I see some other generic sounding classes where conflicts might arise for other users - e.g. `.Stack`, `.App-menu`, `Shape` etc.

**Comment by ad1992** [comment]
> @ad1992 Yep, I meant the opposite case - the bootstrap classes are impacting excalidraw. The scoping implemented in excalidraw works only "one way" - protects others, but not itself.
> 
> > However we can surely add a display class in excalidraw css for this particular use case but there could be more impacting if there is a css outside without scoping.
> 
> This would fix only this specific case, but I think the more general problem of outside classes having an unintended interference with excalidraw would persist.
> 
> The solution I've seen that would fix it would be prefixing all excalidraw classes with some prefix - e.g. `.exc-dropdown-menu` (e.g. all CKEditor classes start with `.ck-`). I see some other generic sounding classes where conflicts might arise for other users - e.g. `.Stack`, `.App-menu`, `Shape` etc.

Yes this sounds good. I think we should do it with all our components to be on safer side. 
@dwelle what do you think?

**Comment by dwelle** [comment]
yeah, we can. I believe this is a duplicate of https://github.com/excalidraw/excalidraw/issues/3825

**Comment by ad1992** [comment]
yep it is. I will close this in favor of #3825
